File: backend/src/main/java/com/realdolmen/backend/domain/Lp.java
package com.realdolmen.backend.domain;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.validation.constraints.Size;

@Entity
@DiscriminatorValue("LP")
public class Lp extends Article{
    @Size(max = 100)
    private String artist;

    @Enumerated(EnumType.STRING)
    private LpGenre genre;

    public String getArtist() {
        return artist;
    }

    public void setArtist(String artist) {
        this.artist = artist;
    }

    public LpGenre getGenre() {
        return genre;
    }

    public void setGenre(LpGenre genre) {
        this.genre = genre;
    }

    @Override
    public String getDisplayValue(){
        return "Lp";
    }
}

File: frontend/angular/src/app/entity/Lp.ts
import {Article} from "./Article";

export class Lp extends Article{

  private _artist: string;
  private _genre: string;

  constructor() {
    super();
  }

  get artist(): string {
    return this._artist;
  }

  set artist(value: string) {
    this._artist = value;
  }

  get genre(): string {
    return this._genre;
  }

  set genre(value: string) {
    this._genre = value;
  }
}
